injestion/external_ingest.py

```
@@ -0,0 +1,109 @@
+                key = str(data["source_id"]).encode("utf-8")
+                value = json.dumps(data).encode("utf-8")
+
+                # Produce to Kafka
+                producer.produce(EXTERNAL_TOPIC, key=key, value=value)
+                producer.flush()
+
+                await websocket.send_text("✅ Published to Kafka")
+            except Exception as e:
+                await websocket.send_text(f"❌ Error: {e}")
+    except WebSocketDisconnect:
+        print(f"Client disconnected")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)